### DISCUSSION TITLE
(Newbie questions) Do I need to update from v 1.8.0? How to update using tarballs?

### DISCUSSION THREAD

**Discussion by firepainting** [rfjakob/gocryptfs#747] [Q&A] [answered]
Hello,

I'm pretty much just a normal user. 

Last week I started using gocryptfs + SiriKali (I'm poor at CLI). Installed both just from the Software Manager (Linux Mint 21.1 Cinnamon) and it is v 1.8.0, released 3 years ago.

Since this is an encryption tool I figured it should be the latest version. Is it still safe to not update it?

My use cases are:
- On Dropbox with daily changes
- Backups on other (multiple) cloud storage providers and HDDs.

For the second part (updating using tarballs), this is embarrassing but I can't figure out what to make of `gocryptfs` and `gocryptfs-xray` (from `gocryptfs_v2.3.2_linux-static_amd64.tar.gz`). Can someone be kind enough to post instructions on how to update, or provide a helpful link, please?

TIA.

**Comment by twiddler** [comment, marked as answer]
`gocryptfs` and `gocryptfs-xray` are binaries, i.e. they are the compiled source code that you can run as a program. See [this wiki page](https://github.com/rfjakob/gocryptfs/wiki/Release-Tarballs#whats-inside-the-linux-static_amd64-tarball).

You can run those binaries by opening a terminal in their directory and typing `./gocryptfs` or `./gocryptfs-xray`. However, common practice is to put them in a directory that your [`PATH` variable](http://www.linfo.org/path_env_var.html) includes. Given that you don't want a program or another user to replace those binaries, you should move them to a directory that's included in `PATH` and can only be modified by a user with super-user priviliges. There are several directories you can choose. I personally have my manually installed binaries in `/usr/local/bin/`.

You can also move the man pages. These are manuals you can view on the terminal by entering `man gocryptfs` and `man gocryptfs-xray`. I keep them at `/usr/local/share/man/man1/`.

You can use the following commands to move the files as described above:

```bash
sudo mv gocryptfs gocryptfs-xray /usr/local/bin
sudo mv gocryptfs.1 gocryptfs-xray.1 /usr/local/share/man/man1
```

This might be a lot of information at once. Take your time to understand the concepts and to get to know your distribution. It'll get easier and eventually fun. 🥳

**Reply by firepainting** [reply]
Thank you so much. Really appreciate it :)
